### ISSUE TITLE
why  unnecessary_statements

### ISSUE THREAD

**Issue by ooooori** [dart-lang/linter#2954]
String get statusString {
    switch (status) {
    case 1:
        return 'abc';
    case 2:
        return 'def';
    case 3:
        return 'xyz';
     default:
         return '';
    }
}

report unnecessary_statements, why?

**Comment by asashour** [comment]
I am unable to reproduce, which Dart SDK version do you use?

```dart
class A {
  int status = 1;

  String get statusString {
    switch (status) {
      case 1:
        return 'abc';
      case 2:
        return 'def';
      case 3:
        return 'xyz';
      default:
        return '';
    }
  }
}
```

**Comment by ooooori** [comment]
> reproduce

Dart  SDK Version: 1.22.0

**Comment by ooooori** [comment]
> I am unable to reproduce, which Dart SDK version do you use?
> 
> ```dart
> class A {
>   int status = 1;
> 
>   String get statusString {
>     switch (status) {
>       case 1:
>         return 'abc';
>       case 2:
>         return 'def';
>       case 3:
>         return 'xyz';
>       default:
>         return '';
>     }
>   }
> }
> ```

Dart SDK Version: 1.22.0

**Comment by pq** [comment]
This one was fixed a while back and if you're able to update to a more recent Flutter release you should get the desired behavior.

(There's tons of new stuff in Flutter 2.5.0 you may enjoy too!  Check out [what's new](https://medium.com/flutter/whats-new-in-flutter-2-5-6f080c3f3dc).)

**Comment by ooooori** [comment]
> This one was fixed a while back and if you're able to update to a more recent Flutter release you should get the desired behavior.
> 
> (There's tons of new stuff in Flutter 2.5.0 you may enjoy too! Check out [what's new](https://medium.com/flutter/whats-new-in-flutter-2-5-6f080c3f3dc).)

Thanks!

**Comment by pq** [comment]
You bet.  Have fun!
